src/cmd_line/commandLine.ts

```diff
@@ -470,4 +506,19 @@ export class SearchCommandLine extends CommandLine {
       vimState
     );
   }
+
+  /**
+   * Called when <C-g> or <C-t> is pressed during SearchInProgress mode
+   */
+  public async advanceCurrentMatch(vimState: VimState, direction: SearchDirection): Promise<void> {
+    // <C-g> always moves forward in the document, and <C-t> always moves back, regardless of search direction.
+    // To compensate, multiply the desired direction by the searchState's direction, so that
+    // effectiveDirection == direction * (searchState.direction)^2 == direction.
+    const originalOffset = this.currentMatchDisplacement;
+    this.currentMatchDisplacement += this.searchState.direction * direction;
+    if (!configuration.wrapscan && !this.getCurrentMatchRange(vimState)) {
```

Comment:
I can't recall why I wrote it that way originally, yours is much better.

Reply:
Oh wait, it's because with `nowrapscan`, if you hit the last match with `<C-g>`, any subsequent `<C-g>` inputs are ignored, and you stay on the last match.  I wrote it that way so that I wouldn't have to duplicate the calculations for determining the correct relative index of the next match.